packages/dappmanager/src/modules/ethClient/ethereumClient.ts

```diff
@@ -104,98 +110,158 @@ export class EthereumClient {
+    const execClientPkg = await packageGet({ dnpName: execClientDnpName });
+
+    const serviceName = this.getServiceNameFromPackage(execClientPkg);
+    const containerName = this.getContainerNameFromService(execClientPkg, serviceName);
+
+    await this.removeContainerAliasFromDockerNetwork({ containerName, fullnodeAlias });
+
+    this.removeFullnodeAliasFromCompose({ execClientDnpName, execClientServiceName: serviceName, fullnodeAlias });
+  }
+
+  private async addAliasToNewExecClient<T extends Network>({ execClientDnpName, fullnodeAlias }: { execClientDnpName: ExecutionClient<T>, fullnodeAlias: string }): Promise<void> {
+    const execClientPkg = await packageGet({ dnpName: execClientDnpName });
+
+    const serviceName = this.getServiceNameFromPackage(execClientPkg);
+    const containerName = this.getContainerNameFromService(execClientPkg, serviceName);
+
+    this.addContainerAliasToDockerNetwork({ containerName, fullnodeAlias });
+
+    this.addFullnodeAliasToCompose({ execClientDnpName, execClientServiceName: serviceName, fullnodeAlias });
+  }
+
+  private getServiceNameFromPackage(pkg: InstalledPackageDetailData): string {
+    return pkg.manifest?.mainService || pkg.containers[0].serviceName;
+  }
+
+  private getContainerNameFromService(pkg: InstalledPackageDetailData, serviceName: string): string {
+    return pkg.containers.find(container => container.serviceName === serviceName)?.containerName || pkg.containers[0].containerName;
+  }
+
+  private async removeContainerAliasFromDockerNetwork({ containerName, fullnodeAlias }: { containerName: string, fullnodeAlias: string }): Promise<void> {
+    const currentEndpointConfig = await getDnCoreNetworkContainerConfig(containerName);
+
+    const updatedAliases = (currentEndpointConfig?.Aliases || []).filter((alias: string) => alias !== fullnodeAlias);
+
     const endpointConfig: Partial<Dockerode.NetworkInfo> = {
       ...currentEndpointConfig,
-      Aliases: [
-        ...(currentEndpointConfig?.Aliases.filter(
-          // according to docs for compose file v3, aliases are declared as strings https://docs.docker.com/compose/compose-file/compose-file-v3/#aliases
-          (item: string) => item !== params.FULLNODE_ALIAS
-        ) || [])
-      ]
+      Aliases: updatedAliases
     };
 
-    if (removeAlias) this.removeFullnodeAliasFromCompose(dnpName, serviceName);
-    else this.addFullnodeAliasToCompose(dnpName, serviceName);
-
-    await dockerNetworkDisconnect(
+    await dockerNetworkReconnect(
       params.DNP_PRIVATE_NETWORK_NAME,
-      previousEthClientContainerName
+      containerName,
+      endpointConfig
     );
-    await dockerNetworkConnect(
+
+  }
+
+  private async addContainerAliasToDockerNetwork({ containerName, fullnodeAlias }: { containerName: string, fullnodeAlias: string }): Promise<void> {
```

Comment:
Same as above here, alias to add does not have to be always the fullnode alias itself, so we could rename "fullnodeAlias" to "aliasToAdd"